src/main/java/net/dv8tion/jda/core/MessageBuilder.java

```diff
@@ -366,18 +431,505 @@ public int getLength()
+        if (fromIndex > endIndex)
+            throw new IndexOutOfBoundsException("fromIndex > endIndex");
+        
+        if (endIndex >= builder.length())
+        {
+            endIndex = builder.length() - 1;
+        }
+
+        int targetCount = target.length();
+        if (targetCount == 0)
+        {
+            return endIndex;
+        }
+
+        int rightIndex = endIndex - targetCount;
+
+        if (fromIndex > rightIndex)
+        {
+            fromIndex = rightIndex;
+        }
+
+        int strLastIndex = targetCount - 1;
+        char strLastChar = target.charAt(strLastIndex);
+
+        int min = fromIndex + targetCount - 1;
+
+        lastCharSearch: for (int i = endIndex; i >= min; i--)
+        {
+            if (builder.charAt(i) == strLastChar)
+            {
+                for (int j = strLastIndex - 1, k = 1; j >= 0; j--, k++)
+                {
+                    if (builder.charAt(i - k) != target.charAt(j))
+                    {
+                        continue lastCharSearch;
+                    }
+                }
+                return i - target.length() + 1;
+            }
+        }
+        return -1;
+    }
+
+    /**
+     * Creates a {@link java.util.Queue Queue} of {@link net.dv8tion.jda.core.entities.Message Message} objects from this MessageBuilder.
+     * <p>
+     * This method splits the content if it exceeds 2000 chars. The spltting behaviour can be customized using {@link SplitPolicy SplitPolicies}.
+     * The method will try the policies in the order they are passed to it.
+     * <p>
+     * This is not MarkDown safe. An easy workaround is to include Zero Witdh Spaces as predetermined breaking points to the message and only split on them.
+     * 
+     * @return the created {@link net.dv8tion.jda.core.entities.Message Messages}
+     */
+    public Queue<Message> buildAll(SplitPolicy... policy)
+    {
+        if (builder.length() == 0)
+            throw new UnsupportedOperationException("Cannot build a Message with no content. (You never added any content to the message)");
+
+        LinkedList<Message> messages = new LinkedList<Message>();
+
+        if (builder.length() <= 2000) {
+            messages.add(this.build());
+            return messages;
+        } 
+
+        if (policy.length == 0)
+        {
+            policy = new SplitPolicy[]{ SplitPolicy.ANYWHERE };
+        }
+        
+        int currentBeginIndex = 0;
+
+        messageLoop: while (currentBeginIndex < builder.length() - 2001)
+        {
+            for (int i = 0; i < policy.length; i++)
+            {
+                try
+                {
+                    int currentEndIndex = policy[i].nextMessage(currentBeginIndex, this);
+                    messages.add(build(currentBeginIndex, currentEndIndex));
+                    currentBeginIndex = currentEndIndex;
+                    continue messageLoop;
+                }
+                catch (Exception e) {}
```

Comment:
`SplitPolicy#nextMessage()` is supposed to throw an exception if finding the nest message's end fails (for example if there's no newline char within a range of 2k chars). looking at it know I think it would be better if it just returns -1 in such a case @DV8FromTheWorld